src/meta/processors/admin/BalanceTask.cpp

```diff
@@ -53,7 +53,10 @@ void BalanceTask::invoke() {
             LOG(INFO) << taskIdStr_ << "Ask the src to give up the leadership.";
             SAVE_STATE();
             if (srcLived_) {
-                client_->transLeader(spaceId_, partId_, src_).thenValue([this](auto&& resp) {
+                // For balance task of single replica, the target leader is src_ itself
+                HostAddr targetLeader = singleReplica_ ? src_ : kRandomPeer;
+                client_->transLeader(spaceId_, partId_, src_, targetLeader)
```

Comment:
if the targetLeader is src_ itself, it is really need call transLeader funciton?

Reply:
In rare case we would call transLeader if src is the target, only when balance data in single replica we would do this to keep code work in same way.